torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -238,6 +241,54 @@ std::string getNcclErrorDetailStr(
   return interpret + err;
 }
 
+control_plane::RegisterHandler dumpHandler{
+    "dump_nccl_trace_pickle",
+    [](const control_plane::Request& req, control_plane::Response& res) {
+      const auto params = req.params();
+      size_t validParamCount = 0;
+
+      // valid params
+      const std::string includeCollectivesStr = "includecollectives";
+      const std::string includeStackTracesStr = "includestacktraces";
+      const std::string onlyActiveStr = "onlyactive";
+
+      std::unordered_map<std::string, bool> expectedParams = {
+          {includeCollectivesStr, true},
+          {includeStackTracesStr, true},
+          {onlyActiveStr, false}};
+
+      for (const auto& [paramName, paramValue] : params) {
+        auto it = expectedParams.find(paramName);
+        if (it != expectedParams.end()) {
```

Comment:
Should we be throwing an error here on unknown params?

Reply:
Yes, we do return a 400 if an `unknown` param is passed in. 
This happens below in `if (validParamCount < param.size())`